src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -74,13 +74,29 @@ def resolve(self, root_reqs, check_supported_wheels):
         requirements = []
         for req in root_reqs:
             if req.constraint:
-                assert req.name
-                assert req.specifier
+                # TODO: Add warnings to accompany these errors, explaining
+                # that these were undocumented behaviour of the old resolver
+                # and will be removed in the new resolver. We need to consider
+                # how we remember to remove these warnings when the new
+                # resolver becomes the default...
```

Comment:
A regular `deprecated(..., gone_in="20.3")` would work IMO.

Reply:
Yeah, but we need to decide when the new resolver will become the default if we do that. And we don't know that yet (we know when we plan on it being available as a non-unstable option, but not when we switch it on by default and drop the old resolver.

Can we do `gone_in="not sure yet"`?

Reply:
You can `gone_in=None`.